manage

```diff
@@ -495,10 +488,9 @@ $app"
       write_list_pid=$!
     fi
     
-    #run manage script for app installs, uninstalls, or updates. Not for file-updates or app refreshes.
-    if [ "$action" == update ] && [[ "$app" =~ ^filelist:.* ]];then #list of files separated by ':'
-      # use the second argument to pass our list of files to update
-      updatable_apps='' updatable_files="$(echo "$app" | sed 's/^filelist://g' | tr ':' '\n')" no_status=true update_now_cli
+    #run manage script for app installs, uninstalls, or updates. Avoid using it for file-updates because that is out of the scope for manage script.
+    if [ "$action" == update-file ];then
+      updatable_apps='' updatable_files="$app" no_status=true update_now_cli
```

Comment:
we have come back full circle. now you have re-implemented the changes I didn't want. Updating one file at a time which duplicates functionality. the `.git` folder will get re-copied each and every update. this folder can get large which will slow down the process unnecessarily.
```
  #delete .git folder, then copy the new one
  rm -rf "${DIRECTORY}/.git" || sudo rm -rf "${DIRECTORY}/.git" || error "Failed to delete old ${DIRECTORY}/.git folder!"
  cp -a "${DIRECTORY}/update/pi-apps/.git" "${DIRECTORY}" || error "Failed to copy new .git folder!"
```

Reply:
Fair point here. But I see the filelist implementation being too clunky compared to its usefulness. Either I could adopt what you did to bundle all app refreshes to apply at once - to detect all update-file lines and push them all through update_now_cli.
Or, I could create a duplicate codepath to avoid using update_now_cli altogether.

Reply:
why not just split `update_now_cli` apart into multiple smaller functions

Reply:
basically this. just like update_app

```
update_file() {  #file name
  local file="$1"
  [ -z "$file" ] && error "update_file(): no file specified!"  
  mkdir -p "$(dirname "${DIRECTORY}/${file}")"
  
  #copy new version
  if cp -f "${DIRECTORY}/update/pi-apps/${file}" "${DIRECTORY}/${file}" ;then
    status_green "${file} file was copied successfully."
  else
    echo -e "\e[91mFailed to copy ${DIRECTORY}/update/pi-apps/${file}! \e[0m"
  fi
}

update_git() {
  #delete .git folder, then copy the new one
  rm -rf "${DIRECTORY}/.git" || sudo rm -rf "${DIRECTORY}/.git" || error "Failed to delete old ${DIRECTORY}/.git folder!"
  cp -a "${DIRECTORY}/update/pi-apps/.git" "${DIRECTORY}" || error "Failed to copy new .git folder!"
}

update_now_cli() { #input: updatable_files and updatable_apps variables
  
  local IFS=$'\n'
  for file in $updatable_files ;do
    update_file "$file"
  done
  
  for app in $updatable_apps ;do
    update_app "$app"
  done
  
  update_git
  
  [ "$no_status" != true ] && status_green "\nPi-Apps updates complete."
  true
}
```

then just call `update_file` directly in manage daemon and at the end call `update_git` if `update-file`, `update`, or `refresh` existed in the `queue`